### ISSUE TITLE
Problem with AbstractModel::getDisginguishedName() for OpenLDAP

### ISSUE THREAD

**Issue by fillup** [Adldap2/Adldap2#278]
Hi there, we found that with OpenLDAP the searching works fine, but if we try to update a record and save it again the query builder loses the DN and then tries to apply the change to the root DSE. After tracing through the code we found the problem is with line https://github.com/Adldap2/Adldap2/blob/master/src/Models/AbstractModel.php#L497

```php
public function getDistinguishedName()
{
    return $this->getAttribute($this->schema->distinguishedName(), 0);
}
```

If we change the second parameter of ```0``` sent to ```getAttribute``` to ```null``` it works fine. The dn attribute apparently is not an array in the way you expect it from AD? I wasn't sure how to fix this without causing problems for AD though.

Perhaps a new configuration option for dnSubKey that defaults to zero but can be set to null if needed could solve the problem?

**Comment by stevebauman** [comment]
Unfortunately the configuration is different for OpenLDAP and ActiveDirectory.

In ActiveDirectory, the distinguished name attribute is located inside an array named `distinguishedname` (as well as a single attribute named `dn`).

This issue is definitely related to this issue here: https://github.com/Adldap2/Adldap2/issues/264

I need to gain more experience with OpenLDAP, as I'm essentially coding blind trying to support it (hence the experimental notice in the requirements section - https://github.com/Adldap2/Adldap2#requirements).

I'll create a local OpenLDAP server and see where I can improve the lacking compatibility.

**Comment by stevebauman** [comment]
This was fixed in [v6.1.2](https://github.com/Adldap2/Adldap2/releases/tag/v6.1.2).
